frontend/src/components/post/Post.js

```diff
@@ -1,9 +1,88 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import styles from './Post.module.css';
+import ReactTimeAgo from 'react-time-ago';
+import TimeAgo from 'javascript-time-ago';
+import en from 'javascript-time-ago/locale/en';
 
-const Post = ({post}) => {
-  return(
-    <article data-cy="post" key={ post._id }>{ post.message }</article>
-  )
-}
+TimeAgo.addLocale(en);
+
+const Post = ({ post, setReload }) => {
+  const user_id = window.localStorage.getItem('user_id');
+  const token = window.localStorage.getItem('token');
+
+  const isPostLikedByUser = post.likes.includes(user_id);
+
+  const [isLiked, toggleIsLiked] = useState(isPostLikedByUser);
+
+  const handleLikeToggle = async () => {
+    toggleIsLiked((likeState) => !likeState);
+    console.log(post);
+    if (user_id) {
+      let url = isLiked ? '/posts/unlike' : '/posts/like';
+      let response = await fetch(url, {
+        method: 'PATCH',
+        headers: {
```

Comment:
We are confused about Line 18 and how it interacts with line 21. To me, it seems that setting isLiked to a function will always mean that isLiked is truthy and thus line 21 will always have posts/unlike as url. If your tests are passing though, clearly this is not the case but is worth discussing.